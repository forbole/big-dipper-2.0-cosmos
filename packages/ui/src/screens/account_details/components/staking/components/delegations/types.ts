import type { DelegationType } from '@/screens/account_details/components/staking/types';

export type ItemType = DelegationType;
