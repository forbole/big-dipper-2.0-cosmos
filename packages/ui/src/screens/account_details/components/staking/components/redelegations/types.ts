import type { RedelegationType } from '@/screens/account_details/components/staking/types';

export type ItemType = RedelegationType;
