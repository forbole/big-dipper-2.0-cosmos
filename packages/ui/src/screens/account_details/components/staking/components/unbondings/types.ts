import type { UnbondingType } from '@/screens/account_details/components/staking/types';

export type ItemType = UnbondingType;
