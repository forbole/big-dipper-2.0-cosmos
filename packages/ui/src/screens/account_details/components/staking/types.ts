export type StakingType<g> = {
  data: {
    [value: number]: g[];
  };
  count: number;
  loading: boolean;
};

export interface DelegationType {
  validator: string;
  amount: TokenUnit;
  reward: TokenUnit;
  commission?: number;
}

export interface RedelegationType {
  from: string;
  to: string;
  amount: TokenUnit;
  completionTime: string;
}

export interface UnbondingType {
  validator: string;
  amount: TokenUnit;
  completionTime: string;
}

export type DelegationsType = StakingType<DelegationType>;
export type RedelegationsType = StakingType<RedelegationType>;
export type UnbondingsType = StakingType<UnbondingType>;

export interface StakingState {
  tab: number;
  delegations: DelegationsType;
  redelegations: RedelegationsType;
  unbondings: UnbondingsType;
}
