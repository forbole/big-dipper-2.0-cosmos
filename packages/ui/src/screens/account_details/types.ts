export interface OverviewType {
  address: string;
  withdrawalAddress: string;
}

export interface BalanceType {
  available: TokenUnit;
  delegate: TokenUnit;
  unbonding: TokenUnit;
  reward: TokenUnit;
  commission?: TokenUnit;
  total: TokenUnit;
}

export interface OtherTokenType {
  denom: string;
  available: TokenUnit;
  reward: TokenUnit;
  commission: TokenUnit;
}

export interface RewardsType {
  [value: string]: TokenUnit;
}

export interface AccountDetailState {
  loading: boolean;
  exists: boolean;
  desmosProfile: DesmosProfile | null;
  overview: OverviewType;
  balance: BalanceType;
  otherTokens: {
    data: OtherTokenType[];
    count: number;
  };
  rewards: RewardsType;
}
