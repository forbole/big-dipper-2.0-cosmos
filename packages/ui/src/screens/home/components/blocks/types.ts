export interface BlockType {
  height: number;
  txs: number;
  timestamp: string;
  proposer: string;
  hash: string;
}

export interface BlocksState {
  loading: boolean;
  items: BlockType[];
}

export type ItemType = BlockType;
