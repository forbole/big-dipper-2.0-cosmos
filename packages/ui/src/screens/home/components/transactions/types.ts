export interface TransactionType {
  height: number;
  hash: string;
  type: string[];
  success: boolean;
  timestamp: string;
  messages: number;
}

export interface TransactionsState {
  loading: boolean;
  items: TransactionType[];
}
