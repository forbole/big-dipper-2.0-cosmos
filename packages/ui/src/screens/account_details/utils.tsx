import {
  AccountBalancesDocument,
  AccountCommissionDocument,
  AccountDelegationBalanceDocument,
  AccountDelegationRewardsDocument,
  AccountUnbondingBalanceDocument,
  AccountWithdrawalAddressDocument,
} from '@/graphql/general/account_details_documents';
import { toValidatorAddress } from '@/utils/prefix_convert';
import axios from 'axios';

import chainConfig from '@/chainConfig';

export const fetchCommission = async (address: string) => {
  const defaultReturnValue = {
    commission: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          validatorAddress: toValidatorAddress(address),
        },
        query: AccountCommissionDocument,
      }
    );
    return data?.data ?? defaultReturnValue;
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchAccountWithdrawalAddress = async (address: string) => {
  const defaultReturnValue = {
    withdrawalAddress: {
      address,
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountWithdrawalAddressDocument,
      }
    );
    return data?.data ?? defaultReturnValue;
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchAvailableBalances = async (address: string) => {
  const defaultReturnValue = {
    accountBalances: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountBalancesDocument,
      }
    );
    return data?.data ?? defaultReturnValue;
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchDelegationBalance = async (address: string) => {
  const defaultReturnValue = {
    delegationBalance: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountDelegationBalanceDocument,
      }
    );
    return data?.data ?? defaultReturnValue;
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchUnbondingBalance = async (address: string) => {
  const defaultReturnValue = {
    unbondingBalance: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountUnbondingBalanceDocument,
      }
    );
    return data?.data ?? defaultReturnValue;
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchRewards = async (address: string) => {
  const defaultReturnValue = {
    delegationRewards: [],
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountDelegationRewardsDocument,
      }
    );
    return data?.data ?? defaultReturnValue;
  } catch (error) {
    return defaultReturnValue;
  }
};
