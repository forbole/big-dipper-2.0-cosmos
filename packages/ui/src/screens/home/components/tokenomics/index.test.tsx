import { TokenomicsDocument } from '@/graphql/types/general_types';
import Tokenomics from '@/screens/home/components/tokenomics';
import { MockTheme, wait } from '@/tests/utils';
import { ApolloClient, ApolloProvider, from, InMemoryCache } from '@apollo/client';
import { MockedProvider } from '@apollo/client/testing';
import renderer from 'react-test-renderer';

// ==================================
// mocks
// ==================================
const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);
jest.mock('@/components/box', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="box" {...props} />
));
// to fix error, this.wrapperNode is null node_modules/recharts/src/component/Tooltip.tsx:143
jest.mock('recharts', () => ({
  ...jest.requireActual('recharts'),
  Tooltip: () => <div id="test-tooltip" />,
}));

const mockTokenomics = jest.fn().mockReturnValue({
  data: {
    stakingParams: [
      {
        params: {
          bond_denom: 'udsm',
          max_entries: 7,
          max_validators: 125,
          unbonding_time: 1209600000000000,
          historical_entries: 10000,
        },
      },
    ],
    stakingPool: [
      {
        bonded: 254578529800,
        unbonded: 204887435198,
      },
    ],
    supply: [
      {
        coins: [
          {
            denom: 'udaric',
            amount: '7987725829900',
          },
          {
            denom: 'upotin',
            amount: '80000000000000',
          },
        ],
      },
    ],
  },
});

// ==================================
// unit tests
// ==================================
describe('screen: Home/Tokenomics', () => {
  it('matches snapshot', async () => {
    const mockClient = new ApolloClient({ link: from([]), cache: new InMemoryCache() });
    let component: renderer.ReactTestRenderer | undefined;

    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <MockedProvider
            mocks={[{ request: { query: TokenomicsDocument }, result: mockTokenomics }]}
          >
            <MockTheme>
              <Tokenomics />
            </MockTheme>
          </MockedProvider>
        </ApolloProvider>
      );
    });
    await wait(renderer.act);

    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
