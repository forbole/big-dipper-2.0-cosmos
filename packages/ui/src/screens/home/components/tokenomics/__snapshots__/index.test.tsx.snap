// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Home/Tokenomics matches snapshot 1`] = `
.emotion-0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  margin: 0;
  font-size: 1.5rem;
  letter-spacing: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 300;
  line-height: 1.2;
  margin-bottom: 16px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 .data__item {
  width: 50%;
  white-space: pre-wrap;
}

.emotion-2 .data__item h4 {
  color: #414141;
}

.emotion-2 .data__item .MuiTypography-caption {
  color: #777777;
}

.emotion-3 {
  margin: 0;
  font-size: 1rem;
  letter-spacing: 0.15px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.235;
}

.emotion-4 {
  margin: 0;
  font-size: 0.75rem;
  letter-spacing: 0.4px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.66;
}

.emotion-7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.emotion-8 .MuiTypography-caption {
  color: #777777;
}

.emotion-8 .legends__item {
  width: 50%;
}

.emotion-8 .legends__item:before {
  content: "";
  display: inline-block;
  width: 12px;
  height: 12px;
  margin-right: 5px;
}

.emotion-8 .legends__item:first-of-type:before {
  background: #2FB6E0;
}

.emotion-8 .legends__item:nth-of-type(2):before {
  background: #FFC93D;
}

.emotion-8 .legends__item:last-child:before {
  background: #20D494;
}

.emotion-8 .legends__item .caption__percent {
  color: #777777;
}

<div
  className="emotion-0"
  id="box"
>
  <h2
    className="MuiTypography-root MuiTypography-h2 emotion-1"
  >
    tokenomics
  </h2>
  <div
    className="emotion-2"
  >
    <div
      className="data__item"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 emotion-3"
      >
        254,578,529,800
         
      </h4>
      <span
        className="MuiTypography-root MuiTypography-caption emotion-4"
      >
        bondedPercent
      </span>
    </div>
    <div
      className="data__item"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 emotion-3"
      >
        -459,465,964,998
         
      </h4>
      <span
        className="MuiTypography-root MuiTypography-caption emotion-4"
      >
        unbondedPercent
      </span>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-8"
    >
      <div
        className="legends__item"
      >
        <span
          className="MuiTypography-root MuiTypography-caption emotion-4"
        >
          bonded
        </span>
      </div>
      <div
        className="legends__item"
      >
        <span
          className="MuiTypography-root MuiTypography-caption emotion-4"
        >
          unbonded
        </span>
      </div>
      <div
        className="legends__item"
      >
        <span
          className="MuiTypography-root MuiTypography-caption emotion-4"
        >
          unbonding
        </span>
      </div>
    </div>
  </div>
</div>
`;
